test/dynamo/test_dynamic_shapes.py

```diff
@@ -114,6 +114,16 @@ def make_dynamic_cls(cls, assume_static_by_default):
     # Cannot call sizes() on tensor with symbolic sizes/strides
 )
 
+unittest.expectedFailure(
+    DynamicShapesMiscTestsDefaultStatic.test_slice_input_dynamic_shapes_static_default
+    # NotImplementedError: SymNodeVariable() is not a constant
+)
+
+unittest.expectedFailure(
+    DynamicShapesMiscTests.test_slice_input_dynamic_shapes
+    # NotImplementedError: SymNodeVariable() is not a constant
+)
```

Comment:
This one is from Avik's PR, see discussion there https://github.com/pytorch/pytorch/pull/95099

Reply:
What we would like to happen here is that we force a guard on the int when it slices into a Python list. But the usage of the int happens deep inside a slice, so it is not obvious at the call site that this is possible. I also tried making as_python_constant work universally for SymNodeVariable, but this causes a ton of fallout (as code now assumes that it can exercise this path.) Probably the right thing to do is add another method for this sort of thing.